keras_cv/metrics/object_detection/box_coco_metrics.py

```diff
@@ -0,0 +1,285 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import sys
+import types
+
+import tensorflow as tf
+import tensorflow.keras as keras
+
+import keras_cv
+from keras_cv import bounding_box
+
+
+class HidePrints:
+    """A basic internal only context manager to hide print statements."""
+
+    def __enter__(self):
+        self._original_stdout = sys.stdout
+        sys.stdout = open(os.devnull, "w")
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        sys.stdout.close()
+        sys.stdout = self._original_stdout
+
+
+def _box_concat(b1, b2):
+    """Concatenates two bounding box batches together."""
+    if b1 is None:
+        return b2
+    if b2 is None:
+        return b1
+
+    result = {}
+    for key in ["boxes", "classes"]:
+        result[key] = tf.concat([b1[key], b2[key]], axis=0)
```

Comment:
I'd expect to see better performance if we keep all the intermediate tensors in a list and only concat them when we evaluate.

(This is what I saw when I tested PyMetric + COCO).

Reply:
Done - thanks for the pointer here.  tf.concat() in eager can be pretty expensive so makes sense to me as the latter approach only runs every eval_steps() epochs.